src/pytest_bdd/scenario.py

```diff
@@ -303,12 +304,29 @@ def scenario_wrapper(request: FixtureRequest, _pytest_bdd_example: dict[str, str
 def collect_example_parametrizations(
     templated_scenario: ScenarioTemplate,
 ) -> list[ParameterSet] | None:
-    if templated_scenario.examples is None:
-        return None
-    if contexts := list(templated_scenario.examples.as_contexts()):
-        return [pytest.param(context, id="-".join(context.values())) for context in contexts]
-    else:
-        return None
+    parametrizations = []
+    has_multiple_examples = len(templated_scenario.examples) > 1
+
+    for example_id, examples in enumerate(templated_scenario.examples):
+        _tags: set = examples.tags or set()
+        example_marks = []
+        if _tags:
+            with warns(PytestUnknownMarkWarning, match=r"Unknown pytest\.mark\.\w+"):
```

Comment:
`warns` asserts that the warning is raised. Why do we want that?

Reply:
My bad - I forgot to replace my original workaround to stop warnings being emitted for unregistered marks!